src/assets/ts/gradesTeacher.ts

```diff
@@ -0,0 +1,198 @@
+import '../css/gradesTeacher.css';
+import axios from '../../axios';
+import {User} from './User';
+import {Grade} from './Grade';
+import {Subject} from './Subject';
+import {TopPanel} from './TopPanel';
+
+//import menu
+TopPanel();
+let subjectList = document.getElementById("subjectList") as HTMLSelectElement;
+let container = document.getElementById("container") as HTMLElement;
+let grades:Array<Grade>;
+subjectList.addEventListener('change',changeSubject)
+
+async function fetchAllData(){
+    await Promise.all([
+        fetchGrades(),
+        fetchSubjects()
+    ])
+     fetchUsers()
+}
+fetchAllData()
+
+// GradeLIST
+async function fetchGrades(){
+    await axios.get('/grades.json')
+    .then(response =>{ 
+            if(response.status != 200){
+        throw new Error("Something went wront while fetching data")
+        }
+    grades = Object.values(response.data) as Array<Grade>;
+    return grades;
+    })
+    .catch( error => {
+        alert (error.message) 
+    })
+}
```

Comment:
axios jest asynchorniczny - czy jest konieczność dopisywania async/await?
A jeżeli tak to czy nie trzeba dopisać tego przy innych funkcjach z wykorzystaniem axios?

Reply:
Wydaje mi się, że jest konieczne. Bez tego raz na kilka załadowań wywala mi błąd, który wynika z tego, że dane z firebase nie zdążą się załadować a wykonują się kolejne funkcje. 
Dodam async/await do pozostałych axiosów.